infra/main.bicep

```diff
@@ -1,120 +1,177 @@
+
+@description('Azure AI Search Resource')
+param azureAISearchName string = '${environmentName}-search-${resourceToken}'
+
+@description('The SKU of the search service you want to create. E.g. free or standard')
+@allowed([
+  'free'
+  'basic'
+  'standard'
+  'standard2'
+  'standard3'
+])
+param azureSearchSku string = 'standard'
 
 @description('Name of Azure OpenAI Resource')
-param OpenAIName string = '${ResourcePrefix}oai'
+param azureOpenAIResourceName string = '${environmentName}-openai-${resourceToken}'
+
+@description('Name of Azure OpenAI Resource SKU')
+param azureOpenAISkuName string = 'S0'
 
 @description('Azure OpenAI GPT Model Deployment Name')
-param OpenAIGPTModel string = 'gpt-35-turbo'
+param azureOpenAIModel string = 'gpt-35-turbo'
 
 @description('Azure OpenAI GPT Model Name')
-param OpenAIGPTModelName string = 'gpt-35-turbo'
+param azureOpenAIModelName string = 'gpt-35-turbo'
 
 @description('Azure OpenAI GPT Model Version')
-param OpenAIGPTModelVersion string = '0613'
+param azureOpenAIModelVersion string = '0613'
 
-@description('Name of Azure AI Search Resource')
-param AISearchName string = '${ResourcePrefix}ais'
+@description('Name of Azure Application Insights Resource')
+param applicationInsightsName string = '${environmentName}-appinsights-${resourceToken}'
 
-@description('Name of Azure Storage Account Resource')
-param StorageAccountName string = '${ResourcePrefix}sa'
+@description('Name of Storage Account')
+param storageAccountName string = 'str${resourceToken}'
 
-@description('Name of Azure Key Vault Resource')
-param KeyVaultName string = '${ResourcePrefix}kv'
+@description('Name of Azure Machine Learning Workspace')
+param machineLearningName string = '${environmentName}-aml-${resourceToken}'
 
-@description('Name of Azure Application Insights Resource')
-param ApplicationInsightsName string = '${ResourcePrefix}ai'
+@description('Id of the user or app to assign application roles')
+param principalId string = ''
 
-@description('Name of Azure Machine Learning Workspace')
-param MachineLearningName string = '${ResourcePrefix}aml'
-
-resource OpenAI 'Microsoft.CognitiveServices/accounts@2023-05-01' = {
-  name: OpenAIName
-  location: Location
-  kind: 'OpenAI'
-  sku: {
-    name: 'S0'
-  }
-  properties: {
+var tags = { 'azd-env-name': environmentName }
+var rgName = 'rg-${environmentName}'
+var keyVaultName = 'kv-${resourceToken}'
 
-  }
+// Organize resources in a resource group
+resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: rgName
+  location: location
+  tags: tags
+}
 
-  resource OpenAIGPTDeployment 'deployments@2023-05-01' = {
-    name: OpenAIGPTModelName
-    properties: {
-      model: {
-        format: 'OpenAI'
-        name: OpenAIGPTModel
-        version: OpenAIGPTModelVersion
-      }
-    }
-    sku: {
-      name: 'Standard'
-      capacity: 30
-    }
+// Store secrets in a keyvault
+module keyvault './shared/keyvault.bicep' = {
```

Comment:
I was going to suggest disabling soft delete for this KV, but then read that we're deprecating that ability soon...so...ignore me

Reply:
`azd down` also asks you if you want to purge